fix: show all data connector password fields

andre-code opened the pull request:
This change introduces the CLOUD_OPTIONS_PROVIDER_OVERRIDE constant, enabling option overrides based on schema and provider. It replaces the need for passing the customisation using the example parameters.

Before / after
![Screenshot 2025-01-10 at 10 37 16](https://github.com/user-attachments/assets/f63e1f63-87f0-470f-9820-9d5637d59d4e)


fix #3476 

/deploy

lorenzo-cavazzi commented on client/src/features/project/components/cloudStorage/AddOrEditCloudStorage.tsx:
Changes should also be reverted in the `help` section to prevent some fields from showing the help text from an example instead of the option.

I think it's enough to swap in `<OptionTruncatedText text={o.help} />` instead of the current
```
<OptionTruncatedText
              text={
                inputType !== "dropdown" && o.filteredExamples?.length > 0
                  ? o.filteredExamples[0]?.help
                  : o.help
              }
            />
```

But give it a try to be sure

![image](https://github.com/user-attachments/assets/c5981f03-6efc-4352-ab73-a63832787031)

lorenzo-cavazzi commented on client/src/features/project/components/cloudStorage/projectCloudStorage.constants.ts:
```diff
@@ -206,6 +174,44 @@ export const CLOUD_OPTIONS_OVERRIDE = {
   },
 } as Record<string, Record<string, Partial<CloudStorageSchemaOptions>>>;
 
+export const CLOUD_OPTIONS_PROVIDER_OVERRIDE = {
```
The solution looks good :+1: 
Could you add an example in the test file? :pray:

lorenzo-cavazzi reviewed (changes requested):
Nice solution! :+1: 
There is another code change necessary to show the correct help text. We can then merge it :rocket:

andre-code commented:
thanks @lorenzo-cavazzi for your review,
I have added a unit test for the new constant and updated the e2e tests to ensure that the password values are displayed correctly. Let me know if there's anything else you'd like me to address!
